internal/utils/net/backendref.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package net
+
+import (
+	"fmt"
+
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+func BackendHostAndPort(backendRef gwapiv1.BackendObjectReference, defaultNamespace string) (string, uint32) {
```

Comment:
can we find a way to reuse https://github.com/envoyproxy/gateway/blob/c943042cb61f5248df6bf9f38d555d4222f65c0d/internal/gatewayapi/route.go#L1068 instead ?

Reply:
let's address this first and then fix it with follow up PR.

Reply:
there is a working implementation using endpoints for the backendRef on the #3062 draft that I'd put together while waiting on APIs to be settled so I could test things out end to end while I had time to focus on it.

might be able to pull pieces of that and use them here, particularly the common bits in the controller that are required to watch the services and endpoints for backendRefs on EnvoyProxies